File: Solutions/Cpp/number_of_1_bits.cpp
/*
    [EASY]
    191. Number of 1 Bits

    Concepts:
    - bit manipulation

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 6.06 MB   [Beats 100%]
*/

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int hWeight = 0;
        for (int bitShift = 0; bitShift < 8*sizeof(uint32_t); bitShift++){
            hWeight += ((n & (1 << bitShift)) >> bitShift);
        }
        return hWeight;
    }
};

File: Solutions/Python/number_of_1_bits.py
"""
    [EASY]
    191. Number of 1 Bits

    Concepts:
    - bit manipulation

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 17.70 MB  [Beats 7.15%]
"""

class Solution:
    def hammingWeight(self, n: int) -> int:
        bit_count = 0
        while n:
            if n & 1:
                bit_count += 1
            n >>= 1

        return bit_count